src/components/structures/auth/Registration.js

```diff
@@ -447,6 +467,7 @@ module.exports = React.createClass({
                 onEditServerDetailsClick={onEditServerDetailsClick}
                 flows={this.state.flows}
                 serverConfig={this.props.serverConfig}
+                canSubmit={this.state.serverIsAlive}
```

Comment:
same here for not preventing form submission without an obvious way to re-check (for registration I guess you could toggle server-type but the user wouldn't know to do that).

Reply:
Registration has a different phases cycle from the other 3, so I'm not convinced I need to fix the form here. There is no button to click if the server is down, regardless of liveliness checks: it's just a spinner.